### PR TITLE
graph/formats/rdf: add Has{All,Any}{Out,In} and Repeat query methods

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

### CODE REVIEW

**Comment by kortschak** [comment]
This needs `AnyOut` and `AnyIn` to be able to provide the power I want. Don't review yet.

**Comment by kortschak** [comment]
You might want to sit down for this one.

File: graph/formats/rdf/query.go [hunk 1]
```diff
@@ -153,6 +232,29 @@ func (q Query) Not(p Query) Query {
...
+// 	}).Result()
+//
+func (q Query) Repeat(fn func(Query) (q Query, ok bool)) Query {
+	for {
+		var ok bool
+		q, ok = fn(q)
+		if !ok || len(q.terms) == 0 {
+			return q
+		}
+	}
```
Comment (kortschak) [inline graph/formats/rdf/query.go:255]: The alternative implementation of `Repeat`,
```
func (q Query) Repeat(fn func(Query) (q Query, ok bool)) Query {
	for {
		r, ok := fn(q)
		if !ok || len(r.terms) == 0 {
			return q
		}
		q = r
	}
}
```
doesn't have this requirement, but I'm not sure which is better. I think being able to obtain an empty result is an important behaviour and that is slightly harder to do in the alternative formulation.

Comment (vladimir-ch) [reply graph/formats/rdf/query.go:255]: > I think being able to obtain an empty result is an important behaviour

Oh, this is very subtle. Why is it important and why the last non-empty result is more interesting/important than the previous non-empty results? Also, `fn` knows what it's returning and so when the repeat ends, so I'd be fine with keeping the current behavior.

**Review by vladimir-ch — APPROVED** [review]

File: graph/formats/rdf/query.go (hunk 1 above)
Comment (vladimir-ch) [reply graph/formats/rdf/query.go:255]: Understood. Let's keep the current behaviour.

**Pull request merged**